lib/scope.js

```diff
@@ -562,7 +562,16 @@ module.exports = {
         set_to: row.value,
       });
       if ( handle ) {
-        fields_to_set.push({ row, handle, tag, type });
+        if ( (tag === 'CANVAS') && row.value === '/sign' ) {
+          // A signature field on a signature page is the only field on that page
+          fields_to_set.push({ row, handle, tag, type });
+          // Get the rest of the variables before...
+          remaining_vars = remaining_vars.concat( var_data.slice( var_data.indexOf(row) + 1 ));
+          // ...stopping the loop so no more /sign rows are gathered
+          break;
+        } else {
+          fields_to_set.push({ row, handle, tag, type });
+        }
```

Comment:
This works for me! This is actually what I originally had, but I was waffling between including this code here or in the next loop.

Reply:
Yeah, it took some experimentation (which you really couldn't do, sorry)